context/MenuContext.tsx

```diff
@@ -0,0 +1,22 @@
+
+import {createContext, useContext, useState} from "react"
+import * as React from 'react'
+
+
+const MenuContext = createContext<any>({})
+export const useMenu = () => useContext(MenuContext)
+export const MenuContextProvider = ({children} : {children:React.ReactNode}) => {
```

Comment:
I'd recommend to start consolidating re-used types to avoid redundancy; I eventually got there working on the backend stuff (see `firebase/documentsDataTypes.ts` for reference).

Here, I'd recommend something along the lines of:
```
// src/typeDefs.ts (or something sensibly named)
export type TContextProviderProps = {
  [children: string]: React.ReactNode
}
```
(and then import into the files in `/context` where appropriate)

Typing objects in TS is really tedious imo, the whole `({ ... } : { ...  }) => { ... }` pattern is a huge eyesore to me, personally, lol, so I tend to typedef these to avoid that...

Not compulsory by any stretch, but figured I'd point it out here anyways since it's a good use case.